core/domain/learner_progress_services_test.py

```diff
@@ -465,6 +470,40 @@ def test_remove_collection_from_incomplete_list(self):
         self.assertEqual(self._get_all_incomplete_collection_ids(
             self.user_id), [])
 
+    def _test_add_private_exploration_activity(self):
+        """Returns the LearnerProgress object containing all the filtered
+        explorations and collections after unpublishing an exploration.
+        """
+        # Unpublish exploration to change status to ACTIVITY_STATUS_PRIVATE.
+        system_user = user_services.UserActionsInfo(feconf.SYSTEM_COMMITTER_ID)
+        rights_manager.unpublish_exploration(system_user, self.EXP_ID_2)
+        private_exploration = exp_services.get_exploration_summary_by_id(
+            self.EXP_ID_2)
+        self.assertEqual(
+            private_exploration.status, constants.ACTIVITY_STATUS_PRIVATE)
+
+        # Call get_activity_progress to get filtered progress.
+        user_activity = learner_progress_services.get_activity_progress(
+            self.user_id)
+        return user_activity[0]
+
+    def _test_add_private_collection_activity(self):
+        """Returns the LearnerProgress object containing all the filtered
+        explorations and collection after unpublishing a collection.
+        """
+        # Unpublish collection to change status to ACTIVITY_STATUS_PRIVATE.
+        system_user = user_services.UserActionsInfo(feconf.SYSTEM_COMMITTER_ID)
+        rights_manager.unpublish_collection(system_user, self.COL_ID_2)
+        private_collection = collection_services.get_collection_summary_by_id(
+            self.COL_ID_2)
+        self.assertEqual(
+            private_collection.status, constants.ACTIVITY_STATUS_PRIVATE)
+
+        # Call get_activity_progress to get filtered progress.
+        user_activity = learner_progress_services.get_activity_progress(
+            self.user_id)
+        return user_activity[0]
```

Comment:
Why is only the first element returned?

Reply:
get_activity_progress returns a list user_activity where:
user_activity[0] = LearnerProgress object which contains:
        filtered_incomplete_exp_summaries,
        filtered_incomplete_collection_summaries,
        filtered_completed_exp_summaries,
        filtered_completed_collection_summaries,
        filtered_exp_playlist_summaries,
        filtered_collection_playlist_summaries
user_activity[1] = map that contains number of nonexistent ids
user_activity[2] = something about completed_to_incomplete_collection_summaries (which is not relevant)

So I only need user_activity[0] to check the filtered list of summaries.